### PR TITLE
Adding Class binding

### PR DESCRIPTION
**Pull request by DevWurm**
Fullfills #89 

The patch makes a binding of the tooltip to classes possible.
The Tooltip reacts to user actions on every element, which is bound by the specified class in the `for` property.

### CODE REVIEW

**Comment by DevWurm** [comment]
I signed it!

File: .gitignore [hunk 1]
```diff
@@ -1 +1,5 @@
+# Dependencies
 bower_components/
+
+# Editor / IDE Configuration
+.idea/
```
Comment (notwaldorf) [inline .gitignore]: Please don't add this to the project `.gitignore` -- this sounds like a local settings for you

**Comment by notwaldorf** [comment]
Oof, this change makes the element super confusing -- won't the same thing be achieved with `manual-mode`, and move this logic in the application, not the element itself?

We tried to model the `for` attribute to match how `label` etc work, and have it point to a specific ID -- I worry that extending this definition to a class might make it confusing to use.

@cdata WDYT?

File: .gitignore (hunk 1 above)
Comment (DevWurm) [inline .gitignore]: Sorry, I wasn't aware of this, I will remove it

**Comment by DevWurm** [comment]
Ok, I see your point, but I think a tooltip is able to provide more general Information and content, which has a wider relevance than a basic label. Therefore it may be usefull if it can be applied to a more general group of DOM elements.

I don't think manual mode is a very nice alternative, because it's pretty tedious to set up all the listeners by hand and makes the code of the tooltip-user more confusing. In addition to that it requires a higher ammount of insight into the tooltip, which leads to a higher ammount of effort than a provided interface for achieving this use case.

Maybe my point gets clearer, when showing a small example from my current project:
The project makes heavy use of data grids.
I want to provide the user a tooltip, which explains them how to sort on columns with this special grid, when they hover over column headers of columns which are sortable.
To achieve this these columns have the class `sortable` and my tooltip is bound to this class. Using classes for binding makes fulfilling this use case a breeze, while using manual mode (or even multiple tooltips) would require additional code and effort.

But I see that providing both selection methods under the same `for` property is confusing and could cause problems, especially if the `for` property is meant to behave like the `for` properties on `label`s. Maybe it would be easier to overlook the API if the selection via class is done by using an explicit property like `for-class`?

**Comment by DevWurm** [comment]
Are there some more opinions about this PR?

**Comment by notwaldorf** [comment]
@DevWurm I'm still not super keen on it. I think having many elements reuse the same tooltip is great, and I'm not arguing against that. I am worried that doing the class-based selector doesn't do that. For example, we will only get the elements with that class at attach time, and if you're adding elements with that class afterwards, they won't trigger the tooltip. I don't think we should recompute the "elements that could be a target" (and possibly set listeners on new ones) every time a tooltip needs to be shown, because that sounds expensive and janky.

Basically, this should either be a tooltip-manager sort of behaviour (that you add on an `x-app` element that contains all the things that could want the same tooltip), or this is logic that a user needs to write (and we can add an example for it, so that at least people can copy paste it)

**Comment by DevWurm** [comment]
@notwaldorf I see, that it would be inconsistent, if the tooltip is not attached to later added elements. But I think, it would be a great feature, if the tooltip would detect newly attached alements in it's targets scope and attach itself to them, if they match a selector. There are native DOM Apis for watching DOM nodes for underlying changes and I don't think, that it would be a very expensive task.

Well, think about the expectations users have to the tooltip: They want an element, which they can attach to an element (or a group of elements) easily and which just displays some defined information. They don't expect to bother around with listening to events and thinking about the logic when to display and to hide their information and how to do this with the API of the tooltip. At least not, when trying to achieve such a simple use case as targeting multiple related elements.

**Comment by DevWurm** [comment]
Seems that there is no  further interest in this.
